### PR TITLE
SUIT-17639 Allow accepting negative numbers to relativePosition comma…

### PR DESCRIPTION
**Pull request by JeKachan**
…nd. Fixing typo in property on generating relativePosition chain.

### CODE REVIEW

File: test/validation/validators.test.js [hunk 1]
```diff
@@ -17,6 +19,27 @@ describe('validators', () => {
 		assert.throws(() => validatePositiveNumber(NaN), /Error/);
 	});
 
+	it('should validate number', () => {
+		assert.strictEqual(validateNumber(0), 0);
```
Comment (polomsky) [inline test/validation/validators.test.js:23]: negative value should be present in the test, that is the purpose of this method, add it to test

File: lib/validation/validators.js [hunk 2]
```diff
@@ -153,18 +153,29 @@ const validatePositiveNumber = (val, name) => {
...
+	return val;
+};
+
+const createStVarOrNumberValidator = (onlyPositiveNumber = false) => (val, name) => {
 	if (!['number', 'string'].includes(typeof val)) {
 		throwError(name + ' should be suitest configuration variable or number');
 	} else if (typeof val === 'number') {
-		validatePositiveNumber(val, name);
+		(onlyPositiveNumber ? validatePositiveNumber : validateNumber)(val, name);
 	} else if (typeof val === 'string' && !/(<%.+%>)/.test(val)) {
```
Comment (polomsky) [inline lib/validation/validators.js:169]: btw, regexp will also accept `A<%var%>B`, this is definitely wrong.
There is also `1<%var%>0` which looks very weird. Please pay attention

**Review by polomsky — CHANGES_REQUESTED** [review]

File: test/validation/validators.test.js (hunk 1 above)
Comment (JeKachan) [reply test/validation/validators.test.js:23]: added

File: lib/validation/validators.js (hunk 2 above)
Comment (JeKachan) [reply lib/validation/validators.js]: fixed

**Review by polomsky — APPROVED** [review]

**Pull request merged**